python_modules/libraries/dagster-duckdb/dagster_duckdb/io_manager.py

```diff
@@ -279,16 +279,23 @@ def get_select_statement(table_slice: TableSlice) -> str:
     @staticmethod
     @contextmanager
     def connect(context, _):
+        config = context.resource_config["connection_config"]
+
+        # support for `custom_user_agent` was added in v1.0.0
+        # https://github.com/duckdb/duckdb/commit/0c66b6007b736ed2197bca54d20c9ad9a5eeef46
+        if duckdb.__version__ > "1.0.0":
```

Comment:
in some of our other libraries we do version checking with `packaging.version`. It seems like that might be a better way to do it than string comparisons? 

Some examples:
https://github.com/dagster-io/dagster/blob/2e2ff40ce339b1f7e8006ea57bebe49e7e4151e1/python_modules/libraries/dagstermill/dagstermill/compat.py#L6-L10

https://github.com/dagster-io/dagster/blob/a35dd3210a20fd728265dac8413a93af31ae40a4/python_modules/libraries/dagster-polars/dagster_polars/io_managers/parquet.py#L71

Reply:
TIL of `packaging.version` thank you!

https://github.com/dagster-io/dagster/pull/23762/commits/199cd090be4757a2b8c8e1b6822263fc42e284bc